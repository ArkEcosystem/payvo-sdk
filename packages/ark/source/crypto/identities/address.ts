import { Base58Check, Hash } from "@payvo/sdk-cryptography";

import { Network } from "../interfaces/networks";
import { MultiSignatureAsset, KeyPair } from "./contracts";
import { PublicKeyError } from "./errors";
import { getPubKeyHash } from "./helpers.js";
import { PublicKey } from "./public-key";

export class Address {
	public static fromPassphrase(passphrase: string, network?: Network): string {
		return Address.fromPublicKey(PublicKey.fromPassphrase(passphrase), network);
	}

	public static fromPublicKey(publicKey: string, network?: Network): string {
		if (!PublicKey.verify(publicKey)) {
			throw new PublicKeyError(publicKey);
		}

		const buffer: Buffer = Hash.ripemd160(Buffer.from(publicKey, "hex"));
		const payload: Buffer = Buffer.alloc(21);

		payload.writeUInt8(getPubKeyHash(network), 0);
		buffer.copy(payload, 1);

		return this.fromBuffer(payload);
	}

	public static fromWIF(wif: string, network?: Network): string {
		return Address.fromPublicKey(PublicKey.fromWIF(wif, network), network);
	}

	public static fromMultiSignatureAsset(asset: MultiSignatureAsset, network?: Network): string {
		return this.fromPublicKey(PublicKey.fromMultiSignatureAsset(asset), network);
	}

	public static fromPrivateKey(privateKey: KeyPair, network?: Network): string {
		return Address.fromPublicKey(privateKey.publicKey, network);
	}

	public static fromBuffer(buffer: Buffer): string {
		return Base58Check.encode(buffer);
	}

	public static toBuffer(
		address: string,
		network?: Network,
	): { addressBuffer: Buffer; addressError?: string } {
		const buffer: Buffer = Base58Check.decode(address);
		const result: { addressBuffer: Buffer; addressError?: string } = {
			addressBuffer: buffer,
		};

		if (buffer[0] !== getPubKeyHash(network)) {
			result.addressError = `Expected address network byte ${getPubKeyHash(network)}, but got ${buffer[0]}.`;
		}

		return result;
	}

	public static validate(address: string): boolean {
		try {
			return Base58Check.decode(address)[0] === getPubKeyHash();
		} catch (err) {
			return false;
		}
	}
}
