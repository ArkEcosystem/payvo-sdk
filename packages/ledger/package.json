{
	"name": "@payvo/sdk-ledger",
	"version": "4.0.0-next.20",
	"description": "Ledger for @payvo/sdk",
	"license": "MIT",
	"contributors": [],
	"type": "module",
	"exports": "./distribution/index.js",
	"types": "./distribution/index.d.ts",
	"files": [
		"/distribution"
	],
	"scripts": {
		"build": "pnpm run clean && tsc",
		"build:analyze": "webpack --config ../../webpack.analyze.cjs",
		"build:release": "webpack --config ../../webpack.config.cjs",
		"build:watch": "pnpm run clean && tsc -w",
		"clean": "rimraf .coverage distribution tmp"
	},
	"dependencies": {
		"@ledgerhq/hw-transport-u2f": "^5.36.0-deprecated",
		"@ledgerhq/hw-transport-webhid": "^6.19.0",
		"@ledgerhq/hw-transport-webusb": "^6.19.0",
		"@payvo/sdk": "workspace:*",
		"platform": "^1.3.6"
	},
	"devDependencies": {
		"@types/node": "^16.11.10"
	},
	"engines": {
		"node": ">=16.0.0"
	},
	"engineStrict": true
}
