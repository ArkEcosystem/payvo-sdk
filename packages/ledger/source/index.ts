// Based on https://github.com/near/near-ledger-js/blob/master/supportedTransports.js

import LedgerU2F from "@ledgerhq/hw-transport-u2f";
import LedgerHID from "@ledgerhq/hw-transport-webhid";
import LedgerUSB from "@ledgerhq/hw-transport-webusb";
import { Services } from "@payvo/sdk";
import platform from "platform";

export class LedgerTransportFactory {
	public static async create(): Promise<Services.LedgerTransport> {
		return new LedgerTransportFactory().#make();
	}

	async #make(): Promise<Services.LedgerTransport> {
		const [supportsHID, supportsUSB, supportsU2F] = await Promise.all([
			this.#supportsHID(),
			this.#supportsUSB(),
			this.#supportsU2F(),
		]);

		if (!supportsHID && !supportsUSB && !supportsU2F) {
			throw new Error("No transports appear to be supported.");
		}

		const supportedTransports: Services.LedgerTransport[] = [];

		if (supportsHID) {
			supportedTransports.push(LedgerHID);
		}

		if (supportsUSB) {
			supportedTransports.push(LedgerUSB);
		}

		if (supportsU2F) {
			supportedTransports.push(LedgerU2F);
		}

		for (const transport of supportedTransports) {
			try {
				return await transport.create();
			} catch (error) {
				if (error.name === "TransportOpenUserCancelled") {
					throw error;
				}
			}
		}

		throw new Error("Failed to connect through any transport.");
	}

	async #supportsHID(): Promise<boolean> {
		try {
			return await LedgerHID.isSupported();
		} catch {
			return false;
		}
	}

	async #supportsUSB(): Promise<boolean> {
		try {
			if (await LedgerUSB.isSupported()) {
				return platform.os.family !== "Windows" && platform.name !== "Opera";
			}

			return false;
		} catch {
			return false;
		}
	}

	async #supportsU2F(): Promise<boolean> {
		try {
			return await LedgerU2F.isSupported();
		} catch {
			return false;
		}
	}
}
