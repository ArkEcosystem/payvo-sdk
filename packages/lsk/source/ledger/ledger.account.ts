import { getPathArray } from "./ledger.helpers.js";

/**
 * Defines an Account to be used when communicating with ledger.
 */
export class LedgerAccount {
	#account = 0;
	#coinIndex = 134; // @TODO get slip44 from manifest

	/**
	 * Specify the account number.
	 * @param {number} newAccount
	 * @returns {this}
	 */
	public account(newAccount: number): this {
		this.#assertValidPath(newAccount);
		this.#account = newAccount;
		return this;
	}

	/**
	 * Derive the path using hardened entries.
	 * @returns {Buffer} defines the path in buffer form.
	 */
	public derivePath(): Buffer {
		const pathArray = getPathArray(`44'/${this.#coinIndex}'/${this.#account}'`);
		const buffer = Buffer.alloc(pathArray.length * 4);

		pathArray.forEach((r, index) => buffer.writeUInt32BE(r, index * 4));

		return buffer;
	}

	/**
	 * Asserts that the given param is a valid path (integer > 0)
	 */
	#assertValidPath(value: number) {
		if (!Number.isInteger(value)) {
			throw new TypeError('Param must be an integer');
		}

		if (value < 0) {
			throw new Error('Param must be greater than zero');
		}
	}
}
