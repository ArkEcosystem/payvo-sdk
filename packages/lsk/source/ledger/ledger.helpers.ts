const crcTable = [
	0x00_00, 0x11_89, 0x23_12, 0x32_9b, 0x46_24, 0x57_ad, 0x65_36, 0x74_bf, 0x8c_48, 0x9d_c1, 0xaf_5a, 0xbe_d3, 0xca_6c,
	0xdb_e5, 0xe9_7e, 0xf8_f7, 0x10_81, 0x01_08, 0x33_93, 0x22_1a, 0x56_a5, 0x47_2c, 0x75_b7, 0x64_3e, 0x9c_c9, 0x8d_40,
	0xbf_db, 0xae_52, 0xda_ed, 0xcb_64, 0xf9_ff, 0xe8_76, 0x21_02, 0x30_8b, 0x02_10, 0x13_99, 0x67_26, 0x76_af, 0x44_34,
	0x55_bd, 0xad_4a, 0xbc_c3, 0x8e_58, 0x9f_d1, 0xeb_6e, 0xfa_e7, 0xc8_7c, 0xd9_f5, 0x31_83, 0x20_0a, 0x12_91, 0x03_18,
	0x77_a7, 0x66_2e, 0x54_b5, 0x45_3c, 0xbd_cb, 0xac_42, 0x9e_d9, 0x8f_50, 0xfb_ef, 0xea_66, 0xd8_fd, 0xc9_74, 0x42_04,
	0x53_8d, 0x61_16, 0x70_9f, 0x04_20, 0x15_a9, 0x27_32, 0x36_bb, 0xce_4c, 0xdf_c5, 0xed_5e, 0xfc_d7, 0x88_68, 0x99_e1,
	0xab_7a, 0xba_f3, 0x52_85, 0x43_0c, 0x71_97, 0x60_1e, 0x14_a1, 0x05_28, 0x37_b3, 0x26_3a, 0xde_cd, 0xcf_44, 0xfd_df,
	0xec_56, 0x98_e9, 0x89_60, 0xbb_fb, 0xaa_72, 0x63_06, 0x72_8f, 0x40_14, 0x51_9d, 0x25_22, 0x34_ab, 0x06_30, 0x17_b9,
	0xef_4e, 0xfe_c7, 0xcc_5c, 0xdd_d5, 0xa9_6a, 0xb8_e3, 0x8a_78, 0x9b_f1, 0x73_87, 0x62_0e, 0x50_95, 0x41_1c, 0x35_a3,
	0x24_2a, 0x16_b1, 0x07_38, 0xff_cf, 0xee_46, 0xdc_dd, 0xcd_54, 0xb9_eb, 0xa8_62, 0x9a_f9, 0x8b_70, 0x84_08, 0x95_81,
	0xa7_1a, 0xb6_93, 0xc2_2c, 0xd3_a5, 0xe1_3e, 0xf0_b7, 0x08_40, 0x19_c9, 0x2b_52, 0x3a_db, 0x4e_64, 0x5f_ed, 0x6d_76,
	0x7c_ff, 0x94_89, 0x85_00, 0xb7_9b, 0xa6_12, 0xd2_ad, 0xc3_24, 0xf1_bf, 0xe0_36, 0x18_c1, 0x09_48, 0x3b_d3, 0x2a_5a,
	0x5e_e5, 0x4f_6c, 0x7d_f7, 0x6c_7e, 0xa5_0a, 0xb4_83, 0x86_18, 0x97_91, 0xe3_2e, 0xf2_a7, 0xc0_3c, 0xd1_b5, 0x29_42,
	0x38_cb, 0x0a_50, 0x1b_d9, 0x6f_66, 0x7e_ef, 0x4c_74, 0x5d_fd, 0xb5_8b, 0xa4_02, 0x96_99, 0x87_10, 0xf3_af, 0xe2_26,
	0xd0_bd, 0xc1_34, 0x39_c3, 0x28_4a, 0x1a_d1, 0x0b_58, 0x7f_e7, 0x6e_6e, 0x5c_f5, 0x4d_7c, 0xc6_0c, 0xd7_85, 0xe5_1e,
	0xf4_97, 0x80_28, 0x91_a1, 0xa3_3a, 0xb2_b3, 0x4a_44, 0x5b_cd, 0x69_56, 0x78_df, 0x0c_60, 0x1d_e9, 0x2f_72, 0x3e_fb,
	0xd6_8d, 0xc7_04, 0xf5_9f, 0xe4_16, 0x90_a9, 0x81_20, 0xb3_bb, 0xa2_32, 0x5a_c5, 0x4b_4c, 0x79_d7, 0x68_5e, 0x1c_e1,
	0x0d_68, 0x3f_f3, 0x2e_7a, 0xe7_0e, 0xf6_87, 0xc4_1c, 0xd5_95, 0xa1_2a, 0xb0_a3, 0x82_38, 0x93_b1, 0x6b_46, 0x7a_cf,
	0x48_54, 0x59_dd, 0x2d_62, 0x3c_eb, 0x0e_70, 0x1f_f9, 0xf7_8f, 0xe6_06, 0xd4_9d, 0xc5_14, 0xb1_ab, 0xa0_22, 0x92_b9,
	0x83_30, 0x7b_c7, 0x6a_4e, 0x58_d5, 0x49_5c, 0x3d_e3, 0x2c_6a, 0x1e_f1, 0x0f_78,
];

const crc16CcittStart = Number.parseInt("FFFF", 16);
const crc = Number.parseInt("FF", 16);
const xorVariable = Number.parseInt("FFFF", 16);

/**
 * Get CRC-16bit-CCITT.
 *
 * @param {Buffer} buffer
 * @returns {int} unsigned CRC 16 bit
 */
export const getCrc16 = (buffer: Buffer): number => {
	let crc16 = crc16CcittStart;

	for (let index = 0; index < buffer.length; index++) {
		crc16 = crcTable[(crc16 ^ buffer.readUInt8(index)) & crc] ^ (crc16 >> 8);
	}

	return (crc16 & xorVariable) ^ xorVariable;
};

export const getPathArray = (text: string): number[] => {
	const HARDENED = 0x80_00_00_00;

	// skip the root
	if (/^m\//i.test(text)) {
		text = text.slice(2);
	}

	const path = text.split("/");
	const result = new Array(path.length);

	for (const [index, element] of path.entries()) {
		const temporary = /(\d+)(['Hh]?)/.exec(element);

		if (temporary === null) {
			throw new Error("Invalid input");
		}

		result[index] = Number.parseInt(temporary[1], 10);

		if (result[index] >= HARDENED) {
			throw new Error("Invalid child index");
		}

		if (temporary[2] === "h" || temporary[2] === "H" || temporary[2] === "'") {
			result[index] += HARDENED;
		} else if (temporary[2].length > 0) {
			throw new Error("Invalid modifier");
		}
	}

	return result;
};
