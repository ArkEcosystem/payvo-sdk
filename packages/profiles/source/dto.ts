export * from "./signed-transaction.dto.js";
export * from "./transaction.collection";
export * from "./transaction.dto.js";
export * from "./transaction.mapper";
