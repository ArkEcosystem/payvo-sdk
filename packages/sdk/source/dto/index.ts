export * from "./confirmed-transaction";
export * from "./signed-transaction";
export * from "./signed-transaction.contract";
export * from "./unspent-transaction";
export * from "./wallet";
