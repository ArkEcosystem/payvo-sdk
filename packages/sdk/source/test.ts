/* istanbul ignore file */

import { randomUUID } from "crypto";
import { ConfigKey, ConfigRepository } from "./coins";
import { Container, BindingType } from "./ioc";
import { NetworkManifest } from "./networks";
import { BigNumberService } from "./services";

const createContainer = ({
	config,
	httpClient,
	manifest,
	meta,
}: {
	config?: ConfigRepository;
	httpClient: any;
	manifest: NetworkManifest;
	meta?: any;
}): Container => {
	config ??= new ConfigRepository({ network: manifest.id, httpClient });

	config.set(ConfigKey.Network, manifest);

	if (meta) {
		for (const [key, value] of Object.entries(meta)) {
			config.set(key, value);
		}
	}

	const container = new Container();
	container.constant(BindingType.ConfigRepository, config);
	container.constant(BindingType.HttpClient, httpClient);
	container.singleton(BindingType.BigNumberService, BigNumberService);

	return container;
};

export const createService = <T = any>({
	config,
	httpClient,
	manifest,
	meta,
	predicate,
	service,
}: {
	config?: ConfigRepository;
	httpClient: any;
	manifest: NetworkManifest;
	meta?: any;
	predicate: any;
	service: any;
}): T => {
	const container = createContainer({
		config,
		httpClient,
		manifest,
		meta,
	});

	if (predicate) {
		predicate(container);
	}

	return container.resolve(service);
};

export const createServiceAsync = async <T = any>({
	config,
	httpClient,
	manifest,
	meta,
	predicate,
	service,
}: {
	config?: ConfigRepository;
	httpClient: any;
	manifest: NetworkManifest;
	meta?: any;
	predicate: any;
	service: any;
}): Promise<T> => {
	const container = createContainer({
		config,
		httpClient,
		manifest,
		meta,
	});

	if (predicate) {
		await predicate(container);
	}

	const uuid: string = randomUUID();

	container.singleton(uuid, service);

	return container.get(uuid);
};
