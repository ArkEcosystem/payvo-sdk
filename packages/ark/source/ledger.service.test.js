import { assert, describe, loader, mockery, test } from "@payvo/sdk-test";

import { Address } from "@arkecosystem/crypto-identities";
import { IoC, Services } from "@payvo/sdk";
import { openTransportReplayer, RecordStore } from "@ledgerhq/hw-transport-mocker";
import nock from "nock";

import { ledger } from "../test/fixtures/ledger";
import { createService } from "../test/mocking";
import { AddressService } from "./address.service";
import { ClientService } from "./client.service";
import { LedgerService } from "./ledger.service";
import { SignedTransactionData } from "./signed-transaction.dto";
import { ConfirmedTransactionData } from "./confirmed-transaction.dto";
import { WalletData } from "./wallet.dto";

const createMockService = async (record) => {
	const transport = await createService(LedgerService, undefined, (container) => {
		container.constant(IoC.BindingType.Container, container);
		container.singleton(IoC.BindingType.AddressService, AddressService);
		container.singleton(IoC.BindingType.ClientService, ClientService);
		container.constant(IoC.BindingType.DataTransferObjects, {
			SignedTransactionData,
			ConfirmedTransactionData,
			WalletData,
		});
		container.singleton(IoC.BindingType.DataTransferObjectService, Services.AbstractDataTransferObjectService);
		container.constant(
			IoC.BindingType.LedgerTransportFactory,
			async () => await openTransportReplayer(RecordStore.fromString(record)),
		);
	});

	await transport.connect();

	return transport;
};

test("should pass with a resolved transport closure", async () => {
	const ark = await createMockService("");

	assert.undefined(await ark.disconnect());
});

test("should pass with an app version", async () => {
	const ark = await createMockService(ledger.appVersion.record);

	assert.is(await ark.getVersion(), ledger.appVersion.result);
});

test("should pass with a compressed publicKey", async () => {
	const ark = await createMockService(ledger.publicKey.record);

	assert.is(await ark.getPublicKey(ledger.bip44.path), ledger.publicKey.result);
});

test("should pass with a compressed publicKey", async () => {
	const ark = await createMockService(ledger.publicKey.record);

	await assert.rejects(() => ark.getExtendedPublicKey(ledger.bip44.path));
});

test("should pass with a schnorr signature", async () => {
	const ark = await createMockService(ledger.transaction.schnorr.record);

	assert.is(
		await ark.signTransaction(ledger.bip44.path, Buffer.from(ledger.transaction.schnorr.payload, "hex")),
		ledger.transaction.schnorr.result,
	);
});

test("should pass with a schnorr signature", async () => {
	const ark = await createMockService(ledger.message.schnorr.record);

	assert.is(
		await ark.signMessage(ledger.bip44.path, Buffer.from(ledger.message.schnorr.payload, "hex")),
		ledger.message.schnorr.result,
	);
});

describe("scan", (suite) => {
	suite.before(() => nock.disableNetConnect());

	suite.after.each(() => nock.cleanAll());

	suite("should scan for legacy wallets", async () => {
		nock(/.+/)
			.get(
				"/api/wallets?address=D9xJncW4ECUSJQWeLP7wncxhDTvNeg2HNK%2CDFgggtreMXQNQKnxHddvkaPHcQbRdK3jyJ%2CDFr1CR81idSmfgQ19KXe4M6keqUEAuU8kF%2CDTYiNbvTKveMtJC8KPPdBrgRWxfPxGp1WV%2CDJyGFrZv4MYKrTMcjzEyhZzdTAJju2Rcjr",
			)
			.reply(200, loader.json(`test/fixtures/client/wallets-page-0.json`))
			.get(
				"/api/wallets?address=DHnV81YdhYDkwCLD8pkxiXh53pGFw435GS%2CDGhLzafzQpBYjDAWP41U4cx5CKZ5BdSnS3%2CDLVXZyKFxLLdyuEtJRUvFoKcorSrnBnq48%2CDFZAfJ1i1LsvhkUk76Piw4v7oTgq12pX9Z%2CDGfNF9bGPss6YKLEqK5gwr4C1M7vgfenzn",
			)
			.reply(200, loader.json(`test/fixtures/client/wallets-page-1.json`));

		const ark = await createMockService(ledger.wallets.record);

		const walletData = await ark.scan({ useLegacy: true });
		assert.length(Object.keys(walletData), 2);
		assert.object(walletData);

		for (const wallet of Object.values(walletData)) {
			const publicKey = wallet.publicKey();

			if (publicKey) {
				assert.is(Address.fromPublicKey(publicKey, { pubKeyHash: 30 }), wallet.address());
			}

			assert.object(wallet.toObject());
		}
	});

	suite("should scan for new wallets", async () => {
		nock(/.+/)
			.get("/api/wallets")
			.query(true)
			.reply(200, loader.json(`test/fixtures/client/wallets-page-0.json`))
			.get("/api/wallets")
			.query(true)
			.reply(200, loader.json(`test/fixtures/client/wallets-page-1.json`));

		const ark = await createMockService(ledger.wallets.record);

		mockery(ark, "getExtendedPublicKey").mockResolvedValue(
			"030fde54605c5d53436217a2849d276376d0b0f12c71219cd62b0a4539e1e75acd",
		);

		const walletData = await ark.scan({ useLegacy: false, startPath: "m/44'/0'/0'/0/0" });
		assert.length(Object.keys(walletData), 1);
		assert.object(walletData);

		for (const wallet of Object.values(walletData)) {
			const publicKey = wallet.publicKey();

			if (publicKey) {
				assert.is(Address.fromPublicKey(publicKey, { pubKeyHash: 30 }), wallet.address());
			}

			assert.object(wallet.toObject());
		}
	});
});

test("#isNanoS", async () => {
	const subject = await createMockService(ledger.message.schnorr.record);

	assert.boolean(await subject.isNanoS());
});

test("#isNanoX", async () => {
	const subject = await createMockService(ledger.message.schnorr.record);

	assert.boolean(await subject.isNanoX());
});

test.run();
