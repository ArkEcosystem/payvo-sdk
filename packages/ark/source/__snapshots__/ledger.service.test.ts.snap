// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scan should scan for legacy wallets 1`] = `
Object {
  "m/44'/1'/0'/0/0": WalletData {
    "bigNumberService": BigNumberService {
      "configRepository": ConfigRepository {},
    },
    "data": Object {
      "address": "D9xJncW4ECUSJQWeLP7wncxhDTvNeg2HNK",
      "attributes": Object {},
      "balance": "15000000000",
      "nonce": "0",
    },
  },
  "m/44'/1'/1'/0/0": WalletData {
    "bigNumberService": BigNumberService {
      "configRepository": ConfigRepository {},
    },
    "data": Object {
      "address": "DFgggtreMXQNQKnxHddvkaPHcQbRdK3jyJ",
      "balance": 0,
      "publicKey": "0348a63f3d9c0050575cbfe70de1722fd73ba205a9d2f7150d0725b99abb331892",
    },
  },
}
`;

exports[`scan should scan for legacy wallets 2`] = `
Object {
  "address": "D9xJncW4ECUSJQWeLP7wncxhDTvNeg2HNK",
  "balance": Object {
    "available": BigNumber {},
    "fees": BigNumber {},
    "total": BigNumber {},
  },
  "isDelegate": false,
  "isMultiSignature": false,
  "isResignedDelegate": false,
  "isSecondSignature": false,
  "nonce": BigNumber {},
  "publicKey": undefined,
  "rank": undefined,
  "username": undefined,
  "votes": undefined,
}
`;

exports[`scan should scan for legacy wallets 3`] = `
Object {
  "address": "DFgggtreMXQNQKnxHddvkaPHcQbRdK3jyJ",
  "balance": Object {
    "available": BigNumber {},
    "fees": BigNumber {},
    "total": BigNumber {},
  },
  "isDelegate": false,
  "isMultiSignature": false,
  "isResignedDelegate": false,
  "isSecondSignature": false,
  "nonce": BigNumber {},
  "publicKey": "0348a63f3d9c0050575cbfe70de1722fd73ba205a9d2f7150d0725b99abb331892",
  "rank": undefined,
  "username": undefined,
  "votes": undefined,
}
`;

exports[`scan should scan for new wallets 1`] = `
Object {
  "m/44'/1'/0'/0/1": WalletData {
    "bigNumberService": BigNumberService {
      "configRepository": ConfigRepository {},
    },
    "data": Object {
      "address": "D9Bv15zRZotbzqfpTMHK6t1ic4YrgRJg3u",
      "balance": 0,
      "publicKey": "03a13ffb326ebef9a8303db65c53582143d7a0f75afe1b75455d4bd880c50c3b54",
    },
  },
}
`;

exports[`scan should scan for new wallets 2`] = `
Object {
  "address": "D9Bv15zRZotbzqfpTMHK6t1ic4YrgRJg3u",
  "balance": Object {
    "available": BigNumber {},
    "fees": BigNumber {},
    "total": BigNumber {},
  },
  "isDelegate": false,
  "isMultiSignature": false,
  "isResignedDelegate": false,
  "isSecondSignature": false,
  "nonce": BigNumber {},
  "publicKey": "03a13ffb326ebef9a8303db65c53582143d7a0f75afe1b75455d4bd880c50c3b54",
  "rank": undefined,
  "username": undefined,
  "votes": undefined,
}
`;
